nautobot/extras/models/groups.py

```diff
@@ -431,3 +446,199 @@ def clean(self):
 
         # Validate `filter` dict
         self.clean_filter()
+
+    def generate_query_for_filter(self, filter_field, value):
+        q = models.Q()
+        if isinstance(filter_field, django_filters.MultipleChoiceFilter):
+            for v in value:
+                q |= models.Q(**filter_field.get_filter_predicate(v))
+        else:
+            q |= filter_field.get_filter_predicate(v)
+        return q
+
+    def generate_query_for_group(self, group):
+        fs = group.filterset_class(group.filter, group.get_queryset())
+
+        q = models.Q()
+
+        for field_name, value in fs.data.items():
+            filter_field = fs.filters[field_name]
+            q &= self.generate_query_for_filter(filter_field, value)
+
+        return q
+
+    def process_group_filters(self, group=None, qs=None):
+        # Start with this group's base queryset
+        base_qs = self.get_queryset()
+        if qs is None:
+            # Clone the base queryset so we can modify it.
+            qs = self.get_queryset()
+
+        if group is None:
+            group = self
+            logger.debug("Processing group %s...", group)
+
+        big_q = models.Q()
+
+        # Enumerate the filters. Recursing into any groups of groups.
+        for membership in group.dynamic_group_memberships.all():
+            group = membership.group
+            logger.debug("Processing group %s...", group)
+            child_memberships = group.dynamic_group_memberships.all()
+            if child_memberships.exists():
+                qs = self.process_group_filters(group=group, qs=qs)
+
+            operator = membership.operator
+
+            logger.debug("%s -> %s -> %s", group, group.filter, operator)
```

Comment:
Would probably be useful to add *some* constant text to this string so as to make grepping and log filtering easier.